### DISCUSSION TITLE
Simple option to update (or otherwise act on; but specifically, update) *all* submodules

### DISCUSSION THREAD

**Discussion by Dumaiu** [magit/magit#4953] [Support] [answered]
Updating all submodules is something I frequently do after switching branches.  Although it's the default behavior for `git submodule update`, and `(magit-submodule-update)` can do this as well with an Emacs prefix argument, it is not directly inaccessible from the Magit submodule transient, AFAICT.

A simple, narrow solution would be a suffix command, bound to, say, `U`.  

A more general solution would be a prefix command, such as `-a`, which would pass an appropriate argument to whatever suffix function `(magit-submodule-...)` followed.

These alternatives are not mutually exclusive.

-----
Re: Magit version 20230206.1143.

**Comment by tarsius** [comment, marked as answer]
As you noticed, you can use a prefix argument to update all initialized modules. When a command is invoked from a transient prefix command, then the prefix arguments go "in between" the prefix command and the suffix command, i.e., `o C-u u`.

**Reply by Dumaiu** [reply]
> When a command is invoked from a transient prefix command, then the prefix arguments go "in between" the prefix command and the suffix command, i.e., o C-u u.

I've used Magit (gratefully) for years and never known this.  If you want to use an Emacs prefix (which you elsewhere call a 'regular prefix') and one or more infixes, which should come first?

> As you noticed, you can use a prefix argument to update all initialized modules.

This I found [documented](https://magit.vc/manual/magit.html#index-o).  However, because I was ignorant of the above, I thought you were referring to an invocation of `magit-submodule-update` as an ordinary Emacs command.

I submit that this general feature would benefit from a more prominent place in the documentation for Magit or Transient, or both.  For instance, there is no 'prefix' entry in the Magit index.  The Transient [FAQ](https://magit.vc/manual/transient/FAQ.html) would also be suitable.  (Unless I'm the only person ever to have to ask about this.  🙄)

**But the most logical place to describe it**, I think, is the Transient [Introduction](https://magit.vc/manual/transient/Introduction.html), where you are already explaining the difference from Transient prefixes; 'If you want to use a regular prefix *with* a Transient command, the regular prefix comes after the Transient prefix but {before|after} any infix...'

Thanks again.

**Reply by kyleam** [reply]
>  If you want to use an Emacs prefix (which you elsewhere call a 'regular prefix') and one or more infixes, which should come first?

The prefix argument should come right before you invoke the suffix command (e.g., `o -r C-u u`)

**Reply by tarsius** [reply]
I've added a note in https://github.com/magit/transient/commit/323d6b6c84e6ae55e578efa1efb6bb4d42a69cac.

**Reply by Dumaiu** [reply]
Nice, thanks.
